delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/syntax/IOSyntax.scala

```diff
@@ -32,6 +32,13 @@ final class IOFunctorOps[E, A, F[_]: Functor](private val io: IO[E, F[A]]) {
 
 final class RejectionOrErrorOps[E: JsonLdEncoder: HttpResponseFields: Encoder, A](private val io: IO[E, A]) {
 
+  /**
+    * Helper method to convert the error channel of the IO to a [[CustomAkkaRejection]] whenever the passed ''filter'' is true.
+    * If the [[PartialFunction]] does not apply, the error channel is left untouched.
+    */
+  def rejectWhen(filter: PartialFunction[E, Boolean]): IO[Response[E], A] =
```

Comment:
why do you need this?

Reply:
it's used in the routes

Reply:
I got confused because you changed `rejectOn` previously. Now I see the difference